plugin.rb

```diff
@@ -344,12 +338,15 @@
     ] = DiscourseActivityPub::ExcerptParser.get_content(post)
   end
   on(:post_edited) do |post, topic_changed, post_revisor|
-    DiscourseActivityPubObject.handle_model_callback(post, :update)
+    post.perform_activity_pub_activity(:update)
   end
   on(:post_created) do |post, opts, user|
-    DiscourseActivityPubObject.handle_model_callback(post, :create)
+    post.perform_activity_pub_activity(:create)
   end
   on(:post_destroyed) do |post, opts, user|
-    DiscourseActivityPubObject.handle_model_callback(post, :delete)
+    post.perform_activity_pub_activity(:delete)
+  end
+  on(:post_recovered) do |post, opts, user|
+    post.perform_activity_pub_activity(:create)
```

Comment:
What stopped me from adding a test is the thought I would just be testing that `on(:post_recovered)` works. The create action is covered in tests. I felt it would be a false positive. But maybe I was wrong.

Reply:
Yeah, true. A test would only ensure that someone in the future doesn't inadvertently delete the `on(:post_recovered)` lines.